cclib/parser/gamessdatparser.py

```diff
@@ -154,31 +147,39 @@ def extract(self, inputfile, line):
 
         if line[1:5] == "$VEC":
 
-            self.nbasis   = None
-            self.nmo      = 0
-            self.mocoeffs = []
+            if not hasattr(self, 'nmo'):
+                self.nmo  = 0
+
+            if not hasattr(self, 'mocoeffs'):
+                self.mocoeffs = []
 
             line = next(inputfile)
 
             while "$END" not in line:
                 
-                moc_line = line.replace('-', ' -').replace('E -', 'E-').strip()
-                mocoeff = [float(m) for m in moc_line.split()[2:]]
-                atom_number = line.split()[0]
-                line_number = moc_line.split()[1]
-
-                if atom_number == str(len(self.mocoeffs)):
-                    self.mocoeffs[-1].extend(mocoeff)
+                mocoeff = [
+                    float(line[0:2]), 
+                    float(line[2:5]), 
+                    float(line[5:5+15]), 
+                    float(line[20:20+15]), 
+                    float(line[35:35+15]), 
+                    float(line[50:50+15]), 
+                    float(line[65:65+15])
+                ]
+                mo_number   = line.split()[0]

```

Comment:
The first two numbers of each of these lines 
```
 1  1 7.23252350E-01 3.22989507E-02 1.04897051E-03-4.40452175E-05 0.00000000E+00
 ```
 don't mean much. The first is the mo number we are currently writing out and the second is just a counter for how many lines we have written out.

I recommend we do something like this:
```
mo_number = int(line[0:2])
line = line[5:]
fixed_width_size = 15
chunks, chunk_size = len(line), len(line)//fixed_width_size
mo_coeff = [ float(line[i:i+chunk_size]) for i in range(0, chunks, chunk_size) ] 
# the entire line is not guaranteed to be there but they are always the same width
if mo_number == str(len(self.mocoeffs)):
    self.extend_attribute("mocoeffs", mocoeff, -1)
```